pallets/liquid-staking/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+            amount));
+    }: {
+        let _ = LiquidStaking::<T>::record_slash(
+            SystemOrigin::Root.into(),
+            agent.clone(),
+            slash_amount
+        );
+    }
+    verify {
+        assert_eq!(pallet_liquid_staking::TotalStakingAsset::<T>::get(), slash_amount);
+        assert_eq!(pallet_liquid_staking::TotalVoucher::<T>::get(), 5_000_000);
+
+        assert_last_event::<T>(pallet_liquid_staking::Event::SlashRecorded(agent, slash_amount).into());
+    }
+
+    unstake {
+        let caller: T::AccountId = whitelisted_caller();
+        initial_set_up::<T>(caller.clone());
+        let amount = 100_000;
+        let unstake_amount = 5_000_000;
+        assert_ok!(LiquidStaking::<T>::stake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            amount));
+    }: {
+        let _ = LiquidStaking::<T>::unstake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            unstake_amount
+        );
+    }
+    verify {
+        assert_eq!(pallet_liquid_staking::TotalStakingAsset::<T>::get(), 0);
+        assert_eq!(pallet_liquid_staking::TotalVoucher::<T>::get(), 0);
+        assert_eq!(
+            <T as LiquidStakingConfig>::Currency::free_balance(CurrencyId::KSM, &caller),
+            INITIAL_AMOUNT - amount
+        );
+        assert_eq!(
+            <T as LiquidStakingConfig>::Currency::free_balance(CurrencyId::KSM, &LiquidStaking::<T>::account_id()),
+            INITIAL_AMOUNT + amount
+        );
+
+        assert_last_event::<T>(pallet_liquid_staking::Event::Unstaked(caller, unstake_amount).into());
+    }
+
+    process_pending_unstake {
+        let caller: T::AccountId = whitelisted_caller();
+        initial_set_up::<T>(caller.clone());
+        let agent: T::AccountId = account("Sample", 6, SEED);
+        let amount = 100_000;
+        let unstake_amount = 5_000_000;
+        assert_ok!(LiquidStaking::<T>::stake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            amount));
+
+        assert_ok!(LiquidStaking::<T>::unstake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            unstake_amount));
+    }: {
+        let _ = LiquidStaking::<T>::process_pending_unstake(
+            SystemOrigin::Root.into(),
+            agent.clone(),
+            caller.clone(),
+            amount
+        );
+    }
+    verify {
+        assert_eq!(pallet_liquid_staking::AccountPendingUnstake::<T>::get(&caller), None,);
+        let processing_unstake = pallet_liquid_staking::AccountProcessingUnstake::<T>::get(&agent, &caller).unwrap();
+        assert_eq!(processing_unstake.len(), 1);
+        assert_eq!(processing_unstake[0].amount, amount);
+        assert_eq!(
+            processing_unstake[0].block_number,
+            frame_system::Pallet::<T>::block_number()
+        );
+
+        assert_last_event::<T>(pallet_liquid_staking::Event::UnstakeProcessing(agent, caller, amount).into());
+    }
+
+    finish_processed_unstake {
```

Comment:
There ma ybe a lot of items in collections, which means we need to benchmark the worst case. Here you just benchmark one stake and one unstake. If the worst case is not able to find, we need to use bounded collection.
Please check this call and above calls.